price-feeder/config/currency_provider_tracker.go

```diff
@@ -111,51 +117,69 @@ func (t *CurrencyProviderTracker) setCoinIDSymbolMap() error {
 	}
 
 	for _, coin := range listResponse {
-		t.coinIDSymbolMap[coin.Symbol] = coin.ID
+		if _, ok := t.coinIDSymbolMap[coin.Symbol]; !ok {
+			t.coinIDSymbolMap[coin.Symbol] = coin.ID
+		}
 	}
 
 	return nil
 }
 
-// setCurrencyProviders queries CoinGecko's tickers endpoint and the osmosis-api assetpairs
-// endpoint to get all the exchanges that support each price feeder currency pair and store
-// it in the CurrencyProviders map.
-func (t *CurrencyProviderTracker) setCurrencyProviders() error {
-	for _, pair := range t.pairs {
-		// check if CoinGecko API supports pair
-		pairBaseID := t.coinIDSymbolMap[strings.ToLower(pair.Base)]
-		coinGeckoResp, err := http.Get(fmt.Sprintf("%s/%s/%s", coinGeckoRestURL, pairBaseID, coinGeckoTickersEndpoint))
-		if err != nil {
-			return err
-		}
-		defer coinGeckoResp.Body.Close()
+// setOsmosisV2CurrencyProviders queries the osmosis-api assetpairs endpoint to get
+func (t *CurrencyProviderTracker) setOsmosisAPIPairs() error {
+	client := &http.Client{
+		Timeout: requestTimeout,
+	}
+	t.osmosisAPIPairs = make(map[string]string)
 
-		var tickerResponse coinTickerResponse
-		if err = json.NewDecoder(coinGeckoResp.Body).Decode(&tickerResponse); err != nil {
+	osmosisResp, err := client.Get(fmt.Sprintf("%s/%s", osmosisV2RestURL, osmosisV2AssetPairsEndpoint))
+	if err != nil {
+		t.logger.Error().Err(err).Msg("Failed to query osmosis api assetpairs endpoint")
+	} else {
```

Comment:
```suggestion
		t.logger.Error().Err(err).Msg("Failed to query osmosis api assetpairs endpoint")
		return err
	} 
```

Reply:
prefer early return, rather than if-else.
Also, need to remove one `}` at the end.

Reply:
Agreed! If else is taken out